packages/core/core/src/services/document-service/webhooks.ts

```diff
@@ -0,0 +1,82 @@
+import { curry } from 'lodash/fp';
+
+import { UID, Schema, Utils, Core, Modules } from '@strapi/types';
+import { sanitize } from '@strapi/utils';
+
+import { getDeepPopulate } from './utils/populate';
+
+const ALLOWED_WEBHOOK_EVENTS = {
+  ENTRY_CREATE: 'entry.create',
+  ENTRY_UPDATE: 'entry.update',
+  ENTRY_DELETE: 'entry.delete',
+  ENTRY_PUBLISH: 'entry.publish',
+  ENTRY_UNPUBLISH: 'entry.unpublish',
+  ENTRY_DRAFT_DISCARD: 'entry.draft-discard',
+};
+
+type WebhookEvent = Utils.Object.Values<typeof ALLOWED_WEBHOOK_EVENTS>;
+
+const sanitizeEntry = async (
+  model: Schema.ContentType<any> | Schema.Component<any>,
+  entry: Modules.Documents.AnyDocument
+) => {
+  return sanitize.sanitizers.defaultSanitizeOutput(
+    {
+      schema: model,
+      getModel(uid) {
+        return strapi.getModel(uid as UID.Schema);
+      },
+    },
+    entry
+  );
+};
+
+/**
+ * Registers the content events that will be emitted using the document service.
+ */
+const registerEntryWebhooks = (strapi: Core.Strapi) => {
```

Comment:
All this code feels strange in the document service 🤔  IMHO the document service should trigger events and never know what a webhook is

Reply:
should we rename "webhooks" to "events" and register those webhooks somewhere else?

Reply:
Yes 👍